vignettes/Making_maps_with_cancensus.Rmd

```diff
@@ -0,0 +1,132 @@
+toronto_geo <- get_census_geometry(dataset='CA16', regions=list(CMA="35535"), 
+                                   level='CSD', geo_format = 'sf')
+
+# or bundled with data at the appropriate geographic level
+toronto_sf <- get_census(dataset='CA16', regions=list(CMA="35535"),
+                      vectors="v_CA16_2397", level='CSD',
+                      geo_format = 'sf', labels = 'short')
+```
+
+You can specify whether you want data returned in an `sf`-class data frame or as a `sp`-class SpatialPolygonsDataFrame object. There are advantages to each spatial format. The `sf` spatial framework is under development as the new principal spatial library for R and takes advantage of the [Simple Features](https://en.wikipedia.org/wiki/Simple_Features) standard. 
+
+On the other hand, `sp` has been around for a long time with development starting in the early 2000s and, as a result, it is a robust and well-maintained package with a large ecosystem of specialized packages that rely on it. First released in October 2016, `sf` is a package still under development can be buggy and prone to breaking. 
+
+**cancensus** retrieves Census geographic data as GeoJSON objects and then converts them into either `sp` or `sf` objects depending on the `geo_format` parameter input. The examples in this vignette assume `sf` class geography.
+
+```{r, eval = FALSE}
+# retrieve sf dataframe
+toronto_sf <- get_census(dataset='CA16', regions=list(CMA="35535"),
+                         vectors="v_CA16_2397", level='CSD',
+                         geo_format = 'sf', labels = 'short')
+```
+
+# Maps with base R graphics
+
+THe `sf` package provides a wrapper around the base R `plot()` function and largely works the same way with most of the same custom graphical parameters. One key thing about `sf.plot()` is that it will plot every column of a sf data frame separately unless you specify a particular colummn. In this case, the household income variable we selected is stored in the `v_CA16_2397` column. Plotting it is easy. 
+
+```{r}
+plot(toronto_sf[14], main = "Toronto Household Income by CSD")
+```
+
+You can specify titles, colour palettes, breaks, background colour, lines, borders, transparency, graticles, and much more by taking advantage of the available parameters. You can also combine layers of graphics objects made with `plot()` by running different plots and adding the `add = TRUE` parameter. 
+
+```{r}
+plot(toronto_sf[14], col = NA, main = "Toronto CSDs with Median HH Income > 100,000", lty = 3)
+plot(toronto_sf[toronto_sf$v_CA16_2397 > 100000,14], col = "red", add = TRUE)
+```
+
+# Maps with ggplot2
+
+Plotting `sf` objects with `ggplot2` is very straightforward with the built-in `geom_sf` layer. However, as of writing, this new geom is only available in the recent dev versions of the `ggplot2` package. The next scheduled CRAN release of `ggplot2` will fully incorporate `geom_sf`. It's easy to install the dev version from Github:
+
+```{r, eval = FALSE}
```

Comment:
I would suggest having both of these lines commented out above the call to `library(ggplot2)` below, rather than in an `eval=FALSE` block, which can be confusing to readers.

Reply:
The reader won't know that it's an `eval=FALSE` block when the vignette is rendered in html, but does it appear in downloaded package documentation?

Reply:
It'll only appear if the user looks at the vignette source. For that reason, I think `eval=FALSE` is to be used extremely sparingly.